scripts/entity/base_entity.gd

```diff
@@ -0,0 +1,22 @@
+class_name BaseEntity extends Node
+## A base entity class which automatically hooks the initial movement and input together.
+
+## The movement component of the entity.
+@export var movement_component: BaseMovementComponent
+## The input component of the entity.
+@export var input_component: BaseInputComponent
+## The health and status component of the entity.
+@export var health_and_status_component: BaseHealthAndStatusComponent
+## The attack components of the entity.
+@export var attack_components: Array[BaseAttackComponent]
```

Comment:
These define more of an "active entity" rather than all entities. This would be good for an "ActiveEntity" like players, enemies, etc.

Reply:
Keeping for now. I think a base entity should just be an ID based on research